src/converters/manual_task_converter.ts

```diff
@@ -0,0 +1,117 @@
+import {IIdentity} from '@essential-projects/iam_contracts';
+
+import {ManualTask, ManualTaskList} from '@process-engine/consumer_api_contracts';
+import {
+  IProcessModelFacade,
+  IProcessModelFacadeFactory,
+  IProcessModelService,
+  Model,
+  Runtime,
+} from '@process-engine/process_engine_contracts';
+/**
+ * Used to cache process models during ManualTask conversion.
+ * This helps to avoid repeated queries against the database for the same ProcessModel.
+ */
+type ProcessModelCache = {[processModelId: string]: Model.Types.Process};
+
+export class ManualTaskConverter {
+
+  private _processModelService: IProcessModelService;
+  private _processModelFacadeFactory: IProcessModelFacadeFactory;
+
+  constructor(processModelService: IProcessModelService,
+              processModelFacadeFactory: IProcessModelFacadeFactory) {
+    this._processModelService = processModelService;
+    this._processModelFacadeFactory = processModelFacadeFactory;
+  }
+
+  private get processModelService(): IProcessModelService {
+    return this._processModelService;
+  }
+
+  private get processModelFacadeFactory(): IProcessModelFacadeFactory {
+    return this._processModelFacadeFactory;
+  }
+
+  public async convertManualTasks(identity: IIdentity, suspendedFlowNodes: Array<Runtime.Types.FlowNodeInstance>): Promise<ManualTaskList> {
+
+    const suspendedManualTasks: Array<ManualTask> = [];
+
+    const processModelCache: ProcessModelCache = {};
+
+    for (const suspendedFlowNode of suspendedFlowNodes) {
+
+      const currentProcessToken: Runtime.Types.ProcessToken = suspendedFlowNode.tokens.find((token: Runtime.Types.ProcessToken): boolean => {
+        return token.type === Runtime.Types.ProcessTokenType.onSuspend;
+      });
+
+      let processModel: Model.Types.Process;
+
+      // To avoid repeated Queries against the database for the same process model,
+      // the retrieved process models will be cached.
+      // So when we want to get a ProcessModel for a ManualTask, we first check the cache and
+      // get the ProcessModel from there.
+      // We only query the database, if the ProcessModel was not yet retrieved.
+      // This avoids timeouts and request-lags when converting large numbers of manual tasks.
+      const cacheHasMatchingEntry: boolean = processModelCache[currentProcessToken.processModelId] !== undefined;
```

Comment:
```suggestion
      const modelInCache: boolean = processModelCache[currentProcessToken.processModelId] !== undefined;
```

is probably a better name. but just a suggestion.